saleor/app/models.py

```diff
@@ -97,10 +98,41 @@ def has_perm(self, perm):
         return perm_value in self.get_permissions()
 
 
+class AppTokenManager(models.Manager):
+    def create_app_token(self, app: App, name: str = "") -> Tuple["AppToken", str]:
+        """Create app token and save the value as hashed password."""
+        token_obj = cast(AppToken, self.model(name=name, app=app))
```

Comment:
Could we generate a token first and then pass everything to `self.model.create(...)` instead of creating an unsaved instance and then immediately changing some of its fields?

Reply:
Yes. I'll change that.